src/main/java/org/gridsuite/study/server/service/shortcircuit/ShortCircuitService.java

```diff
@@ -156,6 +153,41 @@ public String getShortCircuitAnalysisResult(UUID nodeUuid, FaultResultsMode mode
         }
     }
 
+    private String getShortCircuitAnalysisCsvResultResourcePath(UUID nodeUuid, ShortcircuitAnalysisType type) {
+        Optional<UUID> resultUuidOpt = networkModificationTreeService.getComputationResultUuid(nodeUuid,
+                type == ShortcircuitAnalysisType.ALL_BUSES ? ComputationType.SHORT_CIRCUIT : ComputationType.SHORT_CIRCUIT_ONE_BUS);
+        if (resultUuidOpt.isEmpty()) {
+            return null;
+        }
+        String path = DELIMITER + SHORT_CIRCUIT_API_VERSION + "/results/{resultUuid}/csv";
+        return UriComponentsBuilder.fromPath(path).buildAndExpand(resultUuidOpt.get()).toUriString();
+    }
+
+    public byte[] getShortCircuitAnalysisCsvResultResource(URI resourcePath, String headersCsv) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> entity = new HttpEntity<>(headersCsv, headers);
+        ResponseEntity<byte[]> response = restTemplate.exchange(
+                resourcePath,
+                HttpMethod.POST,
+                entity,
+                byte[].class
+        );
+        if (response.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
+            throw new StudyException(SHORT_CIRCUIT_ANALYSIS_NOT_FOUND);
+        }
```

Comment:
Using try catch to catch 4xx and 5xx http errors

try {
            return restTemplate.exchange(resourcePath, HttpMethod.POST, entity, byte[].class).getBody();
} catch (HttpStatusCodeException e) {
            if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
                throw new StudyException(SHORTCIRCUIT_ANALYSIS_NOT_FOUND);
            } else {
                throw handleHttpError(e, SHORTCIRCUIT_ANALYSIS_ERROR);
            }
}

Reply:
done